### PR TITLE
128 emit base64

### PR DESCRIPTION
**Pull request by A3a3e1**

### CODE REVIEW

File: lib/utils/sftpUtils.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+const path = require('path');
+const { arrayBuffer } = require('node:stream/consumers');
+const { MAX_FILE_SIZE } = require('../constants');
+const { unixTimeToIsoDate } = require('./utils');
+
+async function getFileContentInBase64(context, body, dir) {
+  const { logger, client } = context;
+  const filePath = path.join(dir, body.name);
+  const fileSize = body.size;
+  if (fileSize > MAX_FILE_SIZE) {
```
Comment (if0s) [inline lib/utils/sftpUtils.js:11]: Here should be limit 10 MB

File: README.md [hunk 2]
```diff
@@ -169,8 +179,9 @@ Finds a file by criteria in the provided directory and uploads (streams) to the
 * **rights** - (object, required): Rights to file on SFTP server
 * **directory** - (string, required): Directory
 * **path** - (string, required): Full Path
-* **attachment_url** - (string, required): Url to file in storage
-
+* **attachment_url** - (string, required): Url to file in storage. This field will always be empty if the checkbox `Emit file content (Base64) instead of putting it to the storage` is checked
```
Comment (if0s) [inline README.md:182]:
```suggestion
* **attachment_url** - (string, required): Url to file in storage. This field will always be empty if the checkbox `Emit file content (Base64)` is checked
```

File: README.md [hunk 3]
```diff
@@ -101,15 +101,19 @@ none
...
-Finds a file by name in the provided directory and uploads (streams) to the attachment storage
+Finds a file by name in the provided directory and either uploads (streams) its content to the attachment storage or emits its in Base64 representation as a message.
+
 #### Configuration Fields
 * **Allow Empty Result** - (dropdown, optional, defaults to `No`): Do not thrown error when no objects were found
 * **Allow ID to be Omitted** - (dropdown, optional, defaults to `No`): Do not thrown error when object id is missing
+* **Emit file content (Base64)** - (checkbox, optional, defaults to `No`): When checked, a file will not be put into the internal storage. The file content in Base64 representation will be emitted as a part of the message. 
+**Be careful:**
+  * Maximum message size supported on the platform is 10MB. This is why files bigger than 10MB will not be processed.
+  * Retrieve sample of a huge file may freeze the UI.
```
Comment (if0s) [inline README.md:112]: It should be fine now - [Handle large data samples on UI](https://docs.elastic.io/releases/24/05.html#handle-large-data-samples-on-ui)

File: README.md [hunk 4]
```diff
@@ -135,6 +140,11 @@ Finds a file by criteria in the provided directory and uploads (streams) to the
 * **File Upload Retry** - (number, optional, default 5): How many times to retry file upload as attachment to platform storage
 * **Retry Timeout** - (number, optional, default 10000): How long to wait between retry attempts in milliseconds
 * **File Upload Timeout** - (number, optional, default 10000): If a file upload process is longer than the specified number of milliseconds and is not processing any data (receiving or uploading), the timeout will be thrown (the process will be retried if \"File Upload Retry\" set)
+* **Emit file content (Base64)** - (checkbox, optional, defaults to `No`): When checked, a file will not be put into the internal storage. The file content in Base64 representation will be emitted as a part of the message.
+  **Be careful:**
+    * Maximum message size supported on the platform is 10MB. This is why files bigger than 10MB will not be processed.
+    * Retrieve sample of a huge file may freeze the UI.
```
Comment (if0s) [inline README.md:146]: Same here

**Review by if0s — APPROVED** [review]

**Review by HannaTrotsenko — APPROVED** [review]

**Pull request merged**